precompiles/proxy/Proxy.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.0;
+
+/**
+ * @author The Moonbeam Team
+ * @title The interface through which solidity contracts will interact with the Proxy pallet
+ * We follow this same interface including four-byte function selectors, in the precompile that
+ * wraps the pallet
+ * Address :    0x000000000000000000000000000000000000080a
+ */
+
+interface Proxy {
+    /// @dev Defines the proxy permission types that may be combined via `|` operator
+    /// The values start at `0` and are represented as `uint32`
+    enum ProxyType {
+        Any,
+        NonTransfer,
+        Governance,
+        Staking,
+        CancelProxy,
+        Balances,
+        AuthorMapping,
+        IdentityJudgement
+    }
+
+    /// @dev Dispatch the given call from an account that the sender is authorised for through
+    /// addProxy. Removes any corresponding announcement(s)
+    /// @param real the account that the proxy will make a call on behalf of
+    /// @param forceProxyType the exact proxy type to be used and checked for this call, encoded 
+    /// as bytes[] where first byte is either 0x00 for not using the parameter, or `0x01` followed
+    /// by big-endian encoded `uint32` ProxyType
+    /// @param call the call to be made by the real account
+    /// @return A boolean confirming whether the address is a nominator
+    /// Selector: a
+    function proxy(
+        address real, 
+        bytes[] forceProxyType,
+        bytes[] call
+    ) external;
+
+    /// @dev Register a proxy account for the sender that is able to make calls on its behalf
+    /// @param delegate the account that the caller would like to make a proxy
+    /// @param proxyType the permissions allowed for this proxy account
+    /// @param delay the announcement period required of the initial proxy, will generally be zero
+    /// Selector: a
+    function addProxy(
+        address delegate,
+        uint32 proxyType,
```

Comment:
u8 should be enough for proxyType (I doubt we will have more than 256 in the future no?)

Reply:
I was wondering about that, it entirely depends on us `u16` seems more future safe though.